Src/FluentAssertions/Primitives/NegatedStringStartStrategy.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives;
+
+internal class NegatedStringStartStrategy : IStringComparisonStrategy
+{
+    private readonly StringComparison stringComparison;
+
+    public NegatedStringStartStrategy(StringComparison stringComparison)
+    {
+        this.stringComparison = stringComparison;
+    }
+
+    public string ExpectationDescription
+    {
+        get
+        {
+            string predicateDescription = IgnoreCase ? "start with equivalent of" : "start with";
+            return "Expected {context:string} that does not " + predicateDescription + " ";
+        }
+    }
+
+    private bool IgnoreCase
+        => stringComparison == StringComparison.OrdinalIgnoreCase;
+
+    public void ValidateAgainstMismatch(IAssertionScope assertion, string subject, string expected)
+    {
+        bool isMatch = subject.StartsWith(expected, stringComparison);
+
+        if (isMatch)
+        {
+            assertion.FailWith(ExpectationDescription + "{0}{reason}, but found {1}.",
+                expected, subject);
+        }
```

Comment:
Why not just use the internal FA-API?

like

```csharp
assertion
    .ForCondition(subject.StartsWith(expected, stringComparison))
    .FailWith(ExpectationDescription + "{0}{reason}, but found {1}.",
                 expected, subject);
```

Reply:
No reason, I just didn't refactor the existing code :-)
![image](https://github.com/fluentassertions/fluentassertions/assets/3438234/37e445e6-e7fc-40d7-94ac-14962556971c)

Reply:
And that code pre-dates the `ForCondition` construct ;-)